validator-api/src/cache/mod.rs

```diff
@@ -117,12 +127,95 @@ impl ValidatorCache {
             rocket.manage(Self::new()).mount(
                 // this format! is so ugly...
                 format!("/{}", VALIDATOR_API_VERSION),
-                routes![routes::get_mixnodes, routes::get_gateways],
+                routes![
+                    routes::get_mixnodes,
+                    routes::get_gateways,
+                    routes::get_active_mixnodes,
+                    routes::get_active_gateways
+                ],
             )
         })
     }
 
-    async fn update_cache(&self, mixnodes: Vec<MixNodeBond>, gateways: Vec<GatewayBond>) {
+    // TODO: check if all nodes can be compared together,
+    // i.e. they all have the same denom for bonds and delegations
+    fn verify_mixnodes(&self, mixnodes: &[MixNodeBond]) -> bool {
+        if mixnodes.is_empty() {
+            return true;
+        }
+        let expected_denom = &mixnodes[0].bond_amount.denom;
```

Comment:
No action needed now, but in general I think we should formalize denom as an enum at least. Keeping it as string will bite us sooner or later. I've sort of formalized it internally in the tauri-wallet. Formalization would also allow us to trivially convert between denoms so checks like this could go away. We'd also be able to have a sane PartialOrd. Maybe something to discuss during a planning meeting.